include/battle.h

```diff
@@ -793,6 +792,94 @@ struct AiBattleData
     u8 choiceWatch:1;
     u8 padding:6;
 };
+union PACKED StatAnimArg
+{
+    u8 value;
+    u16 value_u16; // So that we can cast from a u16
+    struct PACKED {
+        u8 isNegative:1;
+        u8 harshly:1;
+        u8 stat:6;
+    };
+};
+
+union TRANSPARENT StatFlags
+{
+    int raw;
+    u8 allStats;
+    u32 value;
+    struct {
+        u32 unused:1; // HP
+        u32 attack:1;
+        u32 defense:1;
+        u32 speed:1;
+        u32 spAttack:1;
+        u32 spDefense:1;
+        u32 accuracy:1;
+        u32 evasion:1;
+        u32 padding:24;
+    };
+};
+
+union PACKED StatChangerStringKey
+{
+    u8 value;
+    struct PACKED {
+        u8 skipZeroString:1; // Skips "couldn't go any lower/higher!" strings
+        u8 attack:1;
+        u8 defense:1;
+        u8 speed:1;
+        u8 spAttack:1;
+        u8 spDefense:1;
+        u8 accuracy:1;
+        u8 evasion:1;
+    };
+    struct PACKED {
+        u8 unused:1;
+        u8 allStats:7;
+    };
+};
+
+struct MoveEffectResult
+{
+    // Should be populated at initialisation
+    const u8 *pushInstr; // Instruction that will be pushed if battlescriptPush is set
+    u16 currentMove; // gCurrentMove
+    u16 battlerAbility; // gEffectBattler ability
+    u16 battlerItem; // gEffectBattler held item
+    enum MoveEffects moveEffect:8; // Current move effect
+    enum ItemHoldEffect battlerHoldEffect:8; // gEffectBattler's item hold effect
```

Comment:
```suggestion
    enum ItemHoldEffect holdEffect:8; // gEffectBattler's item hold effect
```
I think just `holdEffect` is fine, that's also most commonly used.